src/core/compact_object_defrag_test.cc

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include <absl/strings/str_cat.h>
+#include <mimalloc.h>
+
+#include <list>
+
+#include "base/gtest.h"
+#include "base/logging.h"
+#include "core/compact_object.h"
+#include "core/mi_memory_resource.h"
+extern "C" {
+#include "redis/redis_aux.h"
+#include "redis/zmalloc.h"  // for non-string objects.
+}
+#include <mimalloc-types.h>
+
+// We must use this file - as using compact_object_test cannot be used
+// due to issue with the standard memory_resource being called to release
+// to make way to the thread local memory manager and this cause a crash
+// due to invalid memory access.
+namespace dfly {
+
+// This code will setup local thread mi malloc allocator
+// The same as we have for the shards
+struct ThreadLocalResoure {
```

Comment:
I tried to make this duplicate the same way we are allocating the memory from the shards down to CompactObj. I did this since the UT was never able to trigger the success call to defrag. I don't think this is really required. But maybe the issue is here?